### PR TITLE
ICU-21249 Adds check of result string for NULL to prevent segmentation fault if

### PR DESCRIPTION
**Pull request by gnrunge**
ICU runs with stubdata only.

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21249
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/cintltst/unumberrangeformattertst.c [hunk 1]
```diff
@@ -67,7 +67,9 @@ static void TestExampleCode() {
     const UChar* str = ufmtval_getString(unumrf_resultAsValue(uresult, &ec), &len, &ec);
     assertSuccessCheck("There should not be a failure in the example code", &ec, TRUE);
     assertUEquals("Should produce expected string result", u"$3 – $5", str);
-    assertIntEquals("Length should be as expected", u_strlen(str), len);
+    if (ec == U_ZERO_ERROR) {
```
Comment (markusicu) [inline icu4c/source/test/cintltst/unumberrangeformattertst.c:70]: We should not check for == zero-error.

Please either test `if (U_SUCCESS(ec)) {` or `if (str != nullptr) {`.

Alternatively, do
```
int32_t resultLength = str != nullptr ? u_strlen(str) : 0;
assertIntEquals("Length should be as expected", resultLength, len);
```

Comment (gnrunge) [reply icu4c/source/test/cintltst/unumberrangeformattertst.c]: Done.

File: icu4c/source/test/cintltst/unumberrangeformattertst.c [hunk 2]
```diff
@@ -65,10 +65,10 @@ static void TestExampleCode() {
     // Get the result string:
     int32_t len;
     const UChar* str = ufmtval_getString(unumrf_resultAsValue(uresult, &ec), &len, &ec);
-    if (assertSuccessCheck("There should not be a failure in the example code", &ec, TRUE)) {
-        assertUEquals("Should produce expected string result", u"$3 – $5", str);
-        assertIntEquals("Length should be as expected", u_strlen(str), len);
-    }
+    assertSuccessCheck("There should not be a failure in the example code", &ec, TRUE);
+    assertUEquals("Should produce expected string result", u"$3 – $5", str);
+    int32_t resultLength = str != NULL ? u_strlen(str) : 0;
```
Comment (markusicu) [inline icu4c/source/test/cintltst/unumberrangeformattertst.c:70]: optional: why the old NULL not the newer nullptr?

**Review by markusicu — APPROVED** [review]
lgtm

It would be nice to retitle the commit and PR to reflect what it's now doing. (`git commit --amend`)

File: icu4c/source/test/cintltst/unumberrangeformattertst.c (hunk 2 above)
Comment (markusicu) [reply icu4c/source/test/cintltst/unumberrangeformattertst.c:70]: oh, wait, it's C not C++. never mind.

**Pull request merged**